lib/ansible/modules/database/mysql/mysql_variables.py

```diff
@@ -177,7 +214,14 @@ def main():
         else:
             module.fail_json(msg="unable to find %s. Exception message: %s" % (config_file, to_native(e)))
 
-    mysqlvar_val = getvariable(cursor, mysqlvar)
+    mysqlvar_val = None
+    if mode == 'global' or mode == 'persist':
+        mysqlvar_val = getvariable(cursor, mysqlvar)
```

Comment:
For `mode == 'persist'`, you have to both check the current value *and* the value in the `mysqld_auto` config.

Reply:
Probably I should check the current value only. I think the key point is to set it just now, at runtime. If it was set before to mysqld-auto.conf, it will just be rewritten.

Reply:
As a user, if `mode == 'persist'`, I expect the module to make sure that it is both globally set, and written to disk. If the module sees it is globally set and won't persist it (and it hasn't been persisted before), that would be a bug for me.

Reply:
seems you're right again.. )

Reply:
i removed a part of tests to reproduce the situation when it is in a desired state in runtime but doesn't have line in mysqld-auto.cnf